### ISSUE TITLE
Improve playing performance with downloaded content

### ISSUE THREAD

**Issue by WolframCrem** [streamyfin/streamyfin#258]
**Describe the solution you'd like**
I was thinking about an toggle that will toggle to play from device (if available) or stream from jellyfin.
That toggle would be located on the page that you see when you select content to watch
- no idea where i will place it or how i will word it.

it would also sync the state to jellyfin as if its streaming from it.

the main reason i want to implement this is because sometimes my connection is bad and a hybrid approach would fix this.

but the idea is that i will implement this feature into an fork and when finished it may get merged.

I'm opening this issue / future request to ask if my idea is alright?
if not what would you change?

**Comment by Simon-Eklundh** [comment]
Here's what I'm reading from this:

You want to add the ability to watch downloaded videos while updating the watch percentage (?) and history(?) as though you were streaming the actual video?

Assuming I've understood what you say correctly: +1 on that :) I'd like that too

**Comment by WolframCrem** [comment]
> Here's what I'm reading from this:
> 
> You want to add the ability to watch downloaded videos while updating the watch percentage (?) and history(?) as though you were streaming the actual video?
> 
> Assuming I've understood what you say correctly: +1 on that :) I'd like that too

Yep that's basically it, was thinking about adding the local file to the video dropdown (maybe icon for local and network streams so there is an separation)

**Comment by Alexk2309** [comment]
@WolframCrem Hey we're working towards this, and this most likely be our next big feature of the app.

**Comment by topiga** [comment]
Hey ! Is it still an issue ?
